spec/models/manageiq/providers/vmware/infra_manager/refresher_storage_profile_spec.rb

```diff
@@ -0,0 +1,107 @@
+require Rails.root.join('spec/tools/vim_data/vim_data_test_helper')
+
+describe ManageIQ::Providers::Vmware::InfraManager::Refresher do
+  before(:each) do
+    _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
+    @ems = FactoryGirl.create(
+      :ems_vmware_with_authentication,
+      :zone => zone, :name => "VC41Test-Prod",
+      :hostname => "VC41Test-Prod.MIQTEST.LOCAL",
+      :ipaddress => "192.168.252.14"
+    )
+
+    allow_any_instance_of(ManageIQ::Providers::Vmware::InfraManager)
+      .to receive(:connect).and_return(FakeMiqVimHandle.new)
+    allow_any_instance_of(ManageIQ::Providers::Vmware::InfraManager)
+      .to receive(:disconnect).and_return(true)
+    allow_any_instance_of(ManageIQ::Providers::Vmware::InfraManager)
+      .to receive(:has_credentials?).and_return(true)
+
+    EmsRefresh.refresh(@ems)
+  end
+
+  context "Handling VMware Storage Profile" do
```

Comment:
Since we've moved to this file, I think this nesting is redundant and can probably be removed.